tests/database_test_utils.py

```diff
@@ -0,0 +1,121 @@
+    for example a status, will be created in the database if the id does not
+    exist yet. We do this to comply with database relationships and make it
+    easier to set up tests with minimal effort.
+    :param specification: a dictionary with a table specification
+    :return: None
+    """
+    if specification["table"] == "task":
+        status_specification = {"id": specification["status_id"], "table": "status"}
+        if not item_exists(status_specification):
+            create_db_entries([status_specification])
+
+
+def get_items(specification: dict[str, str | int]) -> Any:
+    """
+    Create a query based on the dictionary specification and return the result
+    :param specification: a dictionary with a table specification
+    :return: the results of the query
+    """
+    values = ", ".join(
+        key + "=" + str(val) if str(val).isnumeric() else str('"' + val + '"')  # type: ignore
+        for key, val in specification.items()  # type: ignore
+        if key != "table"  # type: ignore
+    )
+    table = specification["table"]
+    statement = f"SELECT * FROM {table} WHERE {values}"  # noqa S608
+    with Session(get_engine()) as session:
+        return session.exec(text(statement)).all()  # type: ignore
+
+
+def item_exists(specification: dict[str, Any]) -> bool:
+    """
+    Check if an item exists in the database with the table and id given
+    in the dictionary
+    :param specification: a dictionary with a table specification
+    :return: True if the item exists in the database, False otherwise
+    """
+    result = get_items(specification)
+    return len(result) != 0
+
+
+def create_db_entries(specifications: list[dict[str, Any]]) -> None:
+    """
+    Given an array of specifications, create the database entries.
+
+    If you want to start with an empty database, use init_db instead!
+
+    See init_db doc for examples.
+
+    :param specifications: an array of dictionaries with table specifications
+    :return: None
+    """
+    for specification in specifications:
+        specification = enrich_with_default_values(specification)
+        exists_specification = {"table": specification["table"], "id": specification["id"]}
+        if not item_exists(exists_specification):
+            fix_missing_relations(specification)
+            table = specification.pop("table")
+            keys = ", ".join(key for key in specification)
+            values = ", ".join(
+                str(val) if str(val).isnumeric() else str("'" + val + "'")
+                for val in specification.values()  # type: ignore
+            )
+            statement = f"INSERT INTO {table} ({keys}) VALUES ({values})"  # noqa S608
+            with Session(get_engine()) as session:
+                session.exec(text(statement))  # type: ignore
+                session.commit()
+
+
+def init_db(specifications: list[dict[str, str | int]] | None = None) -> None:
```

Comment:
This could probably also be a fixture that returns `create_db_entries` (so, the function that can then be called)

All other functions here could moved under this function, so:

```python
@pytest.fixture(scope="module")
def db(specifications: list[dict[str, str | int]] | None = None) -> Callable:
    def create_db_entries(specifications: list[dict[str, Any]]) -> None:
        ...
    ...
    return create_db_entries
```

Reply:
or better, maybe, return a db object that has an init and get item method?

Reply:
@berrydenhartog also suggested that, but if I would make it a fixture, how do I pass the 'specifications' parameter? This is different in many cases and can be quite long, like:
    init_db(
        [
            {"table": "task", "id": 1, "status_id": 1},
            {"table": "task", "id": 2, "status_id": 1},
            {"table": "task", "id": 3, "status_id": 1},
        ]
    )

Reply:
The -> Callable didn't work for me, so I create a class and used a fixture to initiate it at every function call. Does this work for you? @anneschuth